### PR TITLE
Bluetooth: Mesh: Fixes Friend Add Another Update

### LINKED ISSUE #20746: Bluetooth: Mesh: Friend node Adding another Friend Update
Two `friend update` message are required in the `MESH/NODE/FRND/FN/BV-20-C [Adding another Friend Update – Friend Node]`, but I did not find such processing logic in the code. But it's strange that I noticed that I have passed the certification ?

In the authentication process, it is required that the first `friend_update` message should be **MD = 1**,
Only in this case `enqueue_update(frnd, 1);`
I tracked the process: `enqueue_update(frnd, 1); <--- enqueue_friend_pdu;`

However `enqueue_friend_pdu` will not be called at this stage.

If I'm right, the problem is this function `bt_mesh_friend_sec_update`:
https://github.com/zephyrproject-rtos/zephyr/blob/3cc5bda2fa10b924c04dc58ff7055fa7bb683dcc/subsys/bluetooth/mesh/friend.c#L193-L210

If the flag  ` [frnd->sec_update]`is true before we set it, it is obvious that the last `friend_update` will be lost.

### PR DESCRIPTION
**Pull request by LingaoM**
Fixes: #20746

According MESH/NODE/FRND/FN/BV-20-C should add another
Friend Update.

Signed-off-by: Lingao Meng <mengabc1086@gmail.com>

### CODE REVIEW

**Comment by jhedberg** [comment]
@LingoMeng could you fix the build issue? Quoting from Shippable:
```
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c: In function 'update_overwrite':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c:1204:10: error: 'return' with a value, in function returning void [-Werror]
   return false;
          ^~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c:1198:13: note: declared here
 static void update_overwrite(struct net_buf *buf)
             ^~~~~~~~~~~~~~~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c:1217:6: error: 'src' undeclared (first use in this function)
  if (src != net_buf_pull_be16(buf)) {
      ^~~
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c:1217:6: note: each undeclared identifier is reported only once for each function it appears in
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c: In function 'add_update':
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/mesh/friend.c:1243:42: error: expected ';' before '}' token
   update_overwrite((struct net_buf *)cur)
                                          ^
                                          ;
  }                                       
```

File: subsys/bluetooth/mesh/friend.c [hunk 1]
```diff
@@ -1202,6 +1195,56 @@ int bt_mesh_friend_init(void)
...
+	net_buf_skip(buf, 2); /* skip dst */
+
+	if (TRANS_CTL_OP((u8_t *) net_buf_pull_mem(buf, 1))
+			!= TRANS_CTL_OP_FRIEND_UPDATE) {
+		goto end;
+	}
+
+	upd = net_buf_pull_mem(buf, sizeof(*upd));
+	BT_DBG("Update Previous Friend Update MD 0x%02x -> 0x01", upd->md);
+	upd->md = 1;
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/friend.c]: I think this needs to be done in the friend timeout instead. There might be other messages that are enqueued after the security update, which won't be transferred after the changes in this PR. Alternatively, the update in add_update could always go out with MD=1, but it looks like that violates the spec test case.

File: subsys/bluetooth/mesh/friend.c [hunk 2]
```diff
@@ -1202,6 +1195,56 @@ int bt_mesh_friend_init(void)
...
+
+	if (!(net_buf_pull_u8(buf) >> 7)) {
+		goto end;
+	}
+
+	net_buf_pull(buf, 3); /* skip SEQNUM */
+
+	net_buf_skip(buf, 2) /* skip src */
+
+	net_buf_skip(buf, 2); /* skip dst */
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/friend.c]: Combine these three calls into one `net_buf_skip(7)`?

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — CHANGES_REQUESTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/21026#discussion_r352023152

File: subsys/bluetooth/mesh/friend.c [hunk 3]
```diff
@@ -1161,6 +1186,8 @@ static void friend_timeout(struct k_work *work)
 		return;
 	}
 
+	update_overwrite(sys_slist_peek_head(&frnd->queue) != NULL);
```
Comment (trond-snekvik) [inline subsys/bluetooth/mesh/friend.c]: should be `update_overwrite(frnd->last, sys_slist_peek_head(&frnd->queue) != NULL);`

**Review by trond-snekvik — CHANGES_REQUESTED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Comment by joerchan** [comment]
@LingoMeng You have build errors. See shippable.

**Review by jhedberg — APPROVED** [review]

**Pull request merged**